pandas/tests/indexing/test_datetime.py

```diff
@@ -223,7 +223,7 @@ def test_series_partial_set_datetime(self):
                 Timestamp('2011-01-03')]
         exp = Series([np.nan, 0.2, np.nan],
                      index=pd.DatetimeIndex(keys, name='idx'), name='s')
-        tm.assert_series_equal(ser.loc[keys], exp, check_index_type=True)
+        tm.assert_series_equal(ser.reindex(keys), exp, check_index_type=True)
```

Comment:
I find that a bit inconsistent (although it seems explicitly done and commented in the code), but let's leave that discussion out of this PR :-)